src/browser/telemetry.js

```diff
@@ -640,111 +640,148 @@ class Instrumenter {
     this.addListener(
       'dom',
       this._window,
-      'blur',
-      blurHandler,
-      true,
+      ['dragstart', 'dragend', 'dragenter', 'dragleave', 'drop'],
+      (e) => this.handleEvent('dragdrop', e)
     );
+    this.addListener('dom', this._window, ['blur', 'focus'], (e) => this.handleEvent('focus', e));
+    this.addListener('dom', this._window, ['submit', 'invalid'], (e) => this.handleEvent('form', e));
+    this.addListener('dom', this._window, ['input', 'change'], (e) => this.handleEvent('input', e));
+    this.addListener('dom', this._window, ['resize'], (e) => this.handleEvent('resize', e));
   }
 
-  handleClick(evt) {
+  handleEvent(name, evt) {
     try {
-      const e = domUtil.getElementFromEvent(evt, this._document);
-      const hasTag = e && e.tagName;
-      const anchorOrButton =
-        domUtil.isDescribedElement(e, 'a') ||
-        domUtil.isDescribedElement(e, 'button');
-      if (
-        hasTag &&
-        (anchorOrButton ||
-          domUtil.isDescribedElement(e, 'input', ['button', 'submit']))
-      ) {
-        this.captureDomEvent('click', e);
-      } else if (domUtil.isDescribedElement(e, 'input', ['checkbox', 'radio'])) {
-        this.captureDomEvent('input', e, e.value, e.checked);
-      }
+      return {
+        click: this.handleClick,
+        dragdrop: this.handleDrag,
+        focus: this.handleFocus,
+        form: this.handleForm,
+        input: this.handleInput,
+        resize: this.handleResize,
+      }[name].call(this, evt);
     } catch (exc) {
-      // TODO: Not sure what to do here
+      console.log(`${name} handler error`, evt, exc, exc.stack);
     }
   }
 
-  handleBlur(evt) {
-    try {
-      const e = domUtil.getElementFromEvent(evt, this._document);
-      if (e && e.tagName) {
-        if (domUtil.isDescribedElement(e, 'textarea')) {
-          this.captureDomEvent('input', e, e.value);
-        } else if (
-          domUtil.isDescribedElement(e, 'select') &&
-          e.options &&
-          e.options.length
-        ) {
-          this.handleSelectInputChanged(e);
-        } else if (
-          domUtil.isDescribedElement(e, 'input') &&
-          !domUtil.isDescribedElement(e, 'input', [
-            'button',
-            'submit',
-            'hidden',
-            'checkbox',
-            'radio',
-          ])
-        ) {
-          this.captureDomEvent('input', e, e.value);
+  handleClick(evt) {
+    const tagName = evt.target?.tagName.toLowerCase();
+    if (['input', 'select', 'textarea'].includes(tagName)) return;
+
+    this.telemeter.captureClick({
+      type: evt.type,
+      isSynthetic: !evt.isTrusted,
+      element: domUtil.elementString(evt.target),
+      timestamp: _.now(),
+    })
+  }
+
+  handleFocus(evt) {
+    const type = evt.type;
+    const element = evt.target.window ? 'window' : domUtil.elementString(evt.target);
```

Comment:
```suggestion
    const element = evt.target?.window ? 'window' : domUtil.elementString(evt.target);
```
IFF there is a chance `evt.target` _could_ be null.